### PR TITLE
Enable azure tests

### LINKED ISSUE #501: As plugin writer, I can test azure storage

### PR DESCRIPTION
**Pull request by fao89**
closes #501

### CODE REVIEW

File: templates/github/.github/workflows/nightly.yml.j2 [hunk 1]
```diff
@@ -30,6 +30,9 @@ jobs:
           {%- if docs_test %}
           - TEST: docs
           {%- endif %}
+          {%- if test_azure %}
+          - TEST: azure
+          {%- endif %}
```
Comment (fao89) [inline templates/github/.github/workflows/nightly.yml.j2:35]: I've been thinking about only enabling it at nightly job

Comment (mdellweg) [reply templates/github/.github/workflows/nightly.yml.j2:35]: I would say, new tests should run more often. What they are supposed to test is probably a hot area of code change. Once you are confident about them, you can move them to nightly only.

File: templates/github/.github/workflows/scripts/install.sh.j2 [hunk 2]
```diff
@@ -149,6 +149,25 @@ minio_secret_key: "'$MINIO_SECRET_KEY'"' vars/main.yaml
...
+
+if [ "$TEST" = "azure" ]; then
+  mkdir -p azurite
+  cd azurite
+  openssl req -newkey rsa:2048 -x509 -nodes -keyout azkey.pem -new -out azcert.pem -sha256 -days 365 -addext "subjectAltName=DNS:pulp-azurite" -subj "/C=CO/ST=ST/L=LO/O=OR/OU=OU/CN=CN"
+  sudo cp azcert.pem /usr/local/share/ca-certificates/azcert.crt
+  sudo dpkg-reconfigure ca-certificates
+  cd ..
+  sed -i -e '/^services:/a \
+  - name: pulp-azurite\
```
Comment (mdellweg) [inline templates/github/.github/workflows/scripts/install.sh.j2:162]: As if we build our own docker compose. ;)

**Review by mdellweg — COMMENTED** [review]
Looks good so far. I would like to see it in action.

**Comment by fao89** [comment]
> Looks good so far. I would like to see it in action.

https://github.com/pulp/pulp_ansible/pull/649

File: templates/github/.github/workflows/nightly.yml.j2 (hunk 1 above)
Comment (fao89) [reply templates/github/.github/workflows/nightly.yml.j2:35]: I was worried about the CI load, but I agree with you

**Review by mdellweg — APPROVED** [review]

**Pull request merged**